### PR TITLE
Move _pathauto_path_is_callback() to AliasUniquifier::isRoute()

### PR DESCRIPTION
**Pull request by kimpepper**

### CODE REVIEW

**Comment by Berdir** [comment]
I was wondering why you added it to that service, but tracking it down, it's called from implementing our own hook which is invoked in this service.

So we have uniqifuier -> hook -> back to uniquifier.

I'm not a big fan of implementing your own hooks, makes it imho harder to understand code than it has to be. Ther'es also no reason this would have to be alterable or something would want to be called first, and it's not something module specific. What about moving the call directly into the service and only invoke the hook if that  doesn't match?

File: src/AliasUniquifier.php [hunk 1]
```diff
@@ -82,6 +94,12 @@ public function uniquify(&$alias, $source, $langcode) {
    * {@inheritdoc}
    */
   public function isReserved($alias, $source, $langcode = LanguageInterface::LANGCODE_NOT_SPECIFIED) {
+    if ($this->aliasStorageHelper->exists($alias, $source, $langcode)) {
```
Comment (Berdir) [inline src/AliasUniquifier.php:98]: Can we add some inline comments here, mention that we check existing aliases first, then existing routes and then invoke a hook to allow other modules to do something.

File: src/AliasUniquifier.php [hunk 2]
```diff
@@ -102,4 +120,32 @@ public function isReserved($alias, $source, $langcode = LanguageInterface::LANGC
...
+      // alias. Note that we do not need to use is_link() to check for symbolic
+      // links since this returns TRUE for either is_file() or is_dir() already.
+      return TRUE;
+    }
+
+    try {
+      $this->urlMatcher->match('/' . $path);
+      return TRUE;
+    }
+    catch (ResourceNotFoundException $e) {
```
Comment (Berdir) [inline src/AliasUniquifier.php:149]: I dislike using exceptions for the program flow but there is no other way :(

Comment (davereid) [inline src/AliasUniquifier.php:149]: It's what the current code has to do as well, so :shrug:

**Pull request merged**